tsdb/head.go

```diff
@@ -1878,44 +1884,24 @@ func (s *stripeSeries) gc(mint int64, minOOOMmapRef chunks.ChunkDiskMapperRef) (
 		minOOOTime int64 = math.MaxInt64
 	)
 	minMmapFile = math.MaxInt32
+	prevSurvivors := s.survivors.Load().(map[chunks.HeadSeriesRef]*memSeries)
+	survivors := make(map[chunks.HeadSeriesRef]*memSeries, len(prevSurvivors))
```

Comment:
Building a new map here is definitely the most expensive part. It reverts the effects my previous optimization here:

```
goos: darwin
goarch: arm64
pkg: github.com/prometheus/prometheus/tsdb
                            │     main     │              survivors               │
                            │    sec/op    │   sec/op     vs base                 │
Head_Truncate/churn=10-12     125.1m ± 10%   323.3m ± 5%  +158.36% (p=0.000 n=10)
Head_Truncate/churn=100-12    138.9m ± 12%   320.6m ± 7%  +130.81% (p=0.000 n=10)
Head_Truncate/churn=1000-12   136.1m ± 21%   316.9m ± 5%  +132.85% (p=0.000 n=10)
geomean                       133.2m         320.3m       +140.35%

                            │     main     │               survivors               │
                            │     B/op     │     B/op      vs base                 │
Head_Truncate/churn=10-12     28.48Mi ± 0%   67.25Mi ± 0%  +136.17% (p=0.000 n=10)
Head_Truncate/churn=100-12    28.92Mi ± 0%   67.70Mi ± 0%  +134.10% (p=0.000 n=10)
Head_Truncate/churn=1000-12   29.48Mi ± 0%   68.38Mi ± 0%  +131.96% (p=0.000 n=10)
geomean                       28.96Mi        67.78Mi       +134.07%

                            │    main     │              survivors              │
                            │  allocs/op  │  allocs/op   vs base                │
Head_Truncate/churn=10-12     14.98k ± 0%   18.80k ± 0%  +25.53% (p=0.000 n=10)
Head_Truncate/churn=100-12    15.25k ± 0%   19.08k ± 0%  +25.11% (p=0.000 n=10)
Head_Truncate/churn=1000-12   17.71k ± 0%   21.55k ± 0%  +21.71% (p=0.000 n=10)
geomean                       15.93k        19.78k       +24.10%
```